### ISSUE TITLE
issue when rendering yaml with integer numbers

### ISSUE THREAD

**Issue by emilien-puget** [pb33f/libopenapi#270]
github.com/pb33f/libopenapi v0.15.14

i have this snippet of code behaving weirdly, the maximum attribut got rewritten as maximum: !!float 13
no issue when rendering in json, so i guess the issue is on the yaml side, unfortunately i wasnt able to fix the issue myself

```go
			data, err := os.ReadFile(path)
			if err != nil {
				panic(err)
			}

			document, err := libopenapi.NewDocument(data)
			model, _ := document.BuildV3Model()
			bytes, err := model.Model.Render()
			if err != nil {
				panic(err)
			}
			mergedFilePath := filepath.Join(dirPath, "merged", file.Name())
			if err := os.WriteFile(mergedFilePath, bytes, 0644); err != nil {
				panic(err)
			}
```

```yaml
openapi: "3.1.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
components:
  schemas:
    PetMax:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          maximum: 13
          format: int64
        name:
          type: string
        tag:
          type: string

```

yields this
```yaml
openapi: "3.1.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
components:
  schemas:
    PetMax:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          maximum: !!float 13
          format: int64
        name:
          type: string
        tag:
          type: string

```

**Comment by emilien-puget** [comment]
after some debugging steps, i was able to pinpoint the format here
https://github.com/pb33f/libopenapi/blob/cbdaac737460db24e6350bf59b420eddc36cdba3/datamodel/high/node_builder.go#L496

**Comment by emilien-puget** [comment]
to be noted that using a value like ` maximum: 13.1` yields a correct result

**Comment by emilien-puget** [comment]
i may have found a way to fix this https://github.com/pb33f/libopenapi/pull/271

**Comment by daveshanley** [comment]
I love it that you just fix these things. Your contributions are excellent - thank you.

**Comment by daveshanley** [comment]
Fix released in `v0.15.15`, provided by @emilien-puget
